File: 1_Easy/rockpaperscissors/rockpaperscissors.py
# RockPaperScissors
# Difficulty:
# simulates rock paper scissors against the computer

# Steps:
# 1. identify that the computer will represent 1 as rock, 2 as paper, 3 as scissors
# 2. generate a random number from 1 to 3 
# 3. depending on the number, output rock, paper, or scissors

# 4. ask the user to enter in a number 1 to 3 depending on the what they would like to choose

# 5. check to see if the user typed in "1" and the computer chose 1, output tie
# 6. repeat for all of the situations when the user types in "1"

# 7. repeat for all other situations and output tie, win, loss accordingly

# Bonus:
# 1. add a repeat and keep track of how many wins you get
# 2. stop the game when points have reached 3 or -3

import random

points = 0
while True:
  choice = input("Do you want to use 1)rock 2)paper 3)scissors? ")
  
  computer = random.randint(1,3)
  
  if choice=='1' and computer==1:
    print("Player throws rock, computer throws rock, tie!")
  if choice=='1' and computer==2:
    print("Player throws rock, computer throws paper, you lose!")
    points=points-1
  if choice=='1' and computer==3:
    print("Player throws rock, computer throws scissors, you win!")
    points=points+1
  
  if choice=='2' and computer==1:
    print("Player throws paper, computer throws rock, you win!")
    points=points+1
  if choice=='2' and computer==2:
    print("Player throws paper, computer throws paper, tie!")
  if choice=='2' and computer==3:
    print("Player throws paper, computer throws scissors, you lose!")
    points=points-1
  
  if choice=='3' and computer==1:
    print("Player throws scissors, computer throws rock, you lose!")
    points=points-1
  if choice=='3' and computer==2:
    print("Player throws scissors, computer throws paper, you win!")
    points=points+1
  if choice=='3' and computer==3:
    print("Player throws scissors, computer throws scissors, tie!")

  print(points)
  if points==3:
    print("You win the whole game! :)")
    break
  if points==-3:
    print("You lose the whole game :(")
    break

File: webstuff/rockpaperscissors/rockpaperscissors.js
// Rock Paper Scissors
// Play rock paper scissors against the computer

// Steps:
// - generate a random number from 1 to 3 and output it
// - depending on the number, output computer chose rock, paper, scissors

// - ask the user to choose rock paper or scissors
// - check to see if the user chose rock
// - if they did then check what the computer randomly chose
// - output whether the user won loss or tied
// - turn this into a function and call the function if the user chose "rock"

// - repeat for paper and scissors

// Bonus:
// - add score and repeat
// - add point when they win, take point away when they lose

const prompt = require("prompt-sync")();

function cpuOutcome() {
  if (cpu === 1) {
    console.log("CPU chose rock!");
  } else if (cpu === 2) {
    console.log("CPU chose paper!");
  } else {
    console.log("CPU chose scissors!");
  }
}

function rockResults() {
  if (cpu === 1) {
    console.log("CPU chose rock! Tied game!");
  } else if (cpu === 2) {
    console.log("CPU chose paper! You lose!");
  } else {
    console.log("CPU chose scissors! You win!");
  }
}

function paperResults() {
  if (cpu === 1) {
    console.log("CPU chose rock! You win!");
  } else if (cpu === 2) {
    console.log("CPU chose paper! Tied game!");
  } else {
    console.log("CPU chose scissors! You lose!");
  }
}

function scissorResults() {
  if (cpu === 1) {
    console.log("CPU chose rock! You lose!");
  } else if (cpu === 2) {
    console.log("CPU chose paper! You win!");
  } else {
    console.log("CPU chose scissors! Tied game!");
  }
}

let cpu = parseInt(Math.random() * 3 + 1);
let user = prompt("Enter your move: (rock, paper, or scissors) ");

console.log("You chose", user);
cpuOutcome();

if (user === "rock") {
  rockResults();
} else if (user === "paper") {
  paperResults();
} else {
  scissorResults();
}
